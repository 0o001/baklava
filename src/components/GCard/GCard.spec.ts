import { GCard } from './';
import { shallowMount } from '@vue/test-utils';

describe('GCard', () => {
  let wrapper: any;
  const renderWrapper = (options: any = {}) => {
    wrapper = shallowMount(GCard, {
      ...options,
    });
  };

  it('should match snapshot on initial', async () => {
    renderWrapper({
      propsData: {
        title: 'Title',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should have GMiniPagination component if pageable prop is true', () => {
    renderWrapper({
      propsData: {
        pageable: true,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should set currentPage when onPage is triggered', () => {
    renderWrapper();

    wrapper.vm.onPage({ page: 2 });

    expect(wrapper.vm.currentPage).toBe(2);
    expect(wrapper.emitted().page).not.toBe(undefined);
    expect(wrapper.emitted().page[0][0]).toEqual({
      page: 2,
    });
  });

  it('should set GCardPage active data if currentPage is equal the GCardPage page prop', async () => {
    const defaultSlot = [
      {
        componentInstance: undefined,
      },
      {
        componentInstance: {
          active: false,
          page: 1,
        },
      },
      {
        componentInstance: {
          active: false,
          page: 2,
        },
      },
    ];
    renderWrapper({
      propsData: {
        pageable: true,
      },
      computed: {
        defaultSlot: () => defaultSlot,
      },
    });

    wrapper.vm.onPage({ page: 2 });

    await wrapper.vm.$nextTick();

    expect(wrapper.find('gminipagination-stub').props('totalPages')).toBe(2);
    expect(defaultSlot[1].componentInstance?.active).toBe(false);
    expect(defaultSlot[2].componentInstance?.active).toBe(true);
  });

  it('should have message-wrapper if showMessage prop is true', () => {
    renderWrapper({
      propsData: {
        showMessage: true,
        message: 'Message',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
