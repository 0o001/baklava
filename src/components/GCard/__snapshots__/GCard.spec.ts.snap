// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GCard should have GMiniPagination component if pageable prop is true 1`] = `
<gbox-stub
  class="g-card"
  direction="column"
  flex="true"
>
  <gbox-stub
    align="items-center"
    flex="true"
    justify="space-between"
  >
    <gbox-stub>
      <gtext-stub
        color="main-grey-500"
        variant="headline-04"
      >
        
          
        
      </gtext-stub>
    </gbox-stub>
     
    <gbox-stub
      flex="true"
    >
       
      <gminipagination-stub
        totalpages="0"
      />
    </gbox-stub>
  </gbox-stub>
   
  <gbox-stub
    class="body g-mt-15"
  />
</gbox-stub>
`;

exports[`GCard should have message-wrapper if showMessage prop is true 1`] = `
<gbox-stub
  class="g-card"
  direction="column"
  flex="true"
>
  <gbox-stub
    align="items-center"
    flex="true"
    justify="space-between"
  >
    <gbox-stub>
      <gtext-stub
        color="main-grey-500"
        variant="headline-04"
      >
        
          
        
      </gtext-stub>
    </gbox-stub>
     
    <gbox-stub
      flex="true"
    >
       
      <!---->
    </gbox-stub>
  </gbox-stub>
   
  <gbox-stub
    align="items-center"
    class="body g-mt-15"
    direction="column"
    flex="true"
  >
    <gicon-stub
      class="g-mb-15"
      color="red-300"
      fill="none"
      name="alert-circle"
      size="24px"
      stroke="currentColor"
      strokelinecap="round"
      strokelinejoin="round"
      strokewidth="2"
      tag="i"
    />
     
    <gtext-stub
      color="red-200"
      variant="body"
    >
      
        Message
      
    </gtext-stub>
  </gbox-stub>
</gbox-stub>
`;

exports[`GCard should match snapshot on initial 1`] = `
<gbox-stub
  class="g-card"
  direction="column"
  flex="true"
>
  <gbox-stub
    align="items-center"
    flex="true"
    justify="space-between"
  >
    <gbox-stub>
      <gtext-stub
        color="main-grey-500"
        variant="headline-04"
      >
        
          Title
        
      </gtext-stub>
    </gbox-stub>
     
    <gbox-stub
      flex="true"
    >
       
      <!---->
    </gbox-stub>
  </gbox-stub>
   
  <gbox-stub
    class="body g-mt-15"
  />
</gbox-stub>
`;
